cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -59,23 +61,78 @@ const (
 	HintInhibitedForPreDownload Hint = "pre-download"
 )
 
+const hintFilePostfix = "lock"
+
 // HintFile returns the full path of the run inhibition lock file for the given snap.
 func HintFile(snapName string) string {
-	return filepath.Join(InhibitDir, snapName+".lock")
+	return filepath.Join(InhibitDir, fmt.Sprintf("%s.%s", snapName, hintFilePostfix))
+}
+
+func InhibitInfoFile(snapName string, hint Hint) string {
+	return filepath.Join(InhibitDir, fmt.Sprintf("%s.%s", snapName, hint))
 }
 
 func openHintFileLock(snapName string) (*osutil.FileLock, error) {
 	return osutil.NewFileLockWithMode(HintFile(snapName), 0644)
 }
 
-// LockWithHint sets a persistent "snap run" inhibition lock, for the given snap, with a given hint.
+// InhibitInfo holds data of the previous snap revision that will be needed by
+// "snap run" while the snap is unlinked (i.e. the current symlink is removed).
+type InhibitInfo struct {
+	// Revision is the previous revision for the snap being refreshed.
+	Revision snap.Revision `json:"revision"`
+}
+
+func validateInhibitInfo(info InhibitInfo) error {
+	if info.Revision.Unset() {
+		return fmt.Errorf("snap revision cannot be unset")
+	}
+	return nil
+}
+
+func validateHint(hint Hint) error {
+	if len(hint) == 0 {
+		return fmt.Errorf("lock hint cannot be empty")
+	}
+	if string(hint) == hintFilePostfix {
+		return fmt.Errorf("hint cannot have value %q", hintFilePostfix)
+	}
+	return nil
+}
+
+func removeInhibitInfoFiles(snapName string) error {
+	infoGlob := filepath.Join(InhibitDir, snapName+".*")
+	// There should be one file only, but just in case
+	files, err := filepath.Glob(infoGlob)
+	if err != nil {
+		return err
+	}
+	hintFile := fmt.Sprintf("%s.%s", snapName, hintFilePostfix)
```

Comment:
With this removeInhibitInfoFiles() now knows details about how hintfiles are constructed, maybe we could reuse the `HintFile` helper here instead? e.g.
```go
	for _, f := range files {
		// Don't remove hint
		if filepath.Base(f) == filepath.Base(HintFile(snapName)) {
			continue
		}
...
```

Reply:
updated, thanks!